File: JavaFundamentals/03_Arrays/ArraysExercise/src/_08MagicSum.java
import java.util.Arrays;
import java.util.Scanner;

public class _08MagicSum {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int[] numbers = Arrays
                .stream(scanner.nextLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        int magicNumber = Integer.parseInt(scanner.nextLine());

        for (int i = 0; i < numbers.length; i++) {
            int firstNumber = numbers[i];

            for (int j = i + 1; j < numbers.length; j++) {
                int secondNumber = numbers[j];
                int magicSum = firstNumber + secondNumber;

                if (magicSum == magicNumber) {
                    System.out.printf("%d %d%n", firstNumber, secondNumber);
                }
            }
        }
    }
}

File: JavaScriptFundamentals/03_Arrays/03_2_ArraysExercises/08_MagicSum.js
function magicSum(numbers, n) {
    for (let i = 0; i < numbers.length; i++) {
        let firstNum = numbers[i];

        for (let j = i + 1; j < numbers.length; j++) {
            let nextNum = numbers[j];

            if ((firstNum + nextNum) === n) {
                console.log(`${firstNum} ${nextNum}`);
            }
        }
    }
}

magicSum([14, 20, 60, 13, 7, 19, 8], 27);
console.log('----------');
magicSum([1, 2, 3, 4, 5, 6], 6);
console.log('----------');
magicSum([1, 7, 6, 2, 19, 23], 8);
console.log('----------');